platform-manageability-agent/internal/comms/comms.go

```diff
@@ -191,46 +213,130 @@ func (cli *Client) RetrieveActivationDetails(ctx context.Context, hostID string,
 		// TODO:
 		// This is a placeholder, replace with actual logic to fetch the password.
 		// Need to check how to fetch the password from dm-manager, hardcoded for now.
-		password := "P@ssw0rd"
+		password := resp.ActionPassword
 		output, err := cli.Executor.ExecuteAMTActivate(rpsAddress, resp.ProfileName, password)
 		if err != nil {
 			return fmt.Errorf("failed to execute activation command for host %s: %w, Output: %s",
 				hostID, err, string(output))
 		}
 		log.Logger.Debugf("Activation command output for host %s: %s", hostID, string(output))
 
-		var req *pb.ActivationResultRequest
-		if isProvisioned(string(output)) {
-			req = &pb.ActivationResultRequest{
-				HostId:           hostID,
-				ActivationStatus: pb.ActivationStatus_PROVISIONED,
-			}
-			log.Logger.Infof("Provisioning successful for host: %s", hostID)
-		} else {
-			req = &pb.ActivationResultRequest{
-				HostId:           hostID,
-				ActivationStatus: pb.ActivationStatus_FAILED,
-			}
-			log.Logger.Infof("Provisioning failed for host: %s", hostID)
+		// Monitor RAS Remote Status with 3-minute timeout
+		activationStatus, err := cli.monitorActivationStatus(ctx, hostID)
+		if err != nil {
+			return fmt.Errorf("failed to monitor activation status for host %s: %w", hostID, err)
+		}
+
+		req := &pb.ActivationResultRequest{
+			HostId:           hostID,
+			ActivationStatus: activationStatus,
 		}
+
 		_, err = cli.DMMgrClient.ReportActivationResults(ctx, req)
 		if err != nil {
-			return fmt.Errorf("failed to report Activation results for host: %s, error: %w", hostID, err)
+			return fmt.Errorf("failed to report activation results for host %s: %w", hostID, err)
 		}
 		log.Logger.Debugf("Reported activation results: HostID=%s, ActivationStatus=%v",
 			hostID, req.ActivationStatus)
 	}
 	return nil
 }
 
-// isProvisioned checks if the output contains the line indicating provisioning success.
-func isProvisioned(output string) bool {
-	scanner := bufio.NewScanner(strings.NewReader(output))
-	for scanner.Scan() {
-		line := scanner.Text()
-		if strings.Contains(line, `msg="CIRA: Configured"`) {
-			return true
+// monitorActivationStatus monitors the RAS Remote Status for up to 3 minutes
+func (cli *Client) monitorActivationStatus(ctx context.Context, hostID string) (pb.ActivationStatus, error) {
+	const (
+		monitorTimeout = 3 * time.Minute
```

Comment:
Instead of running a separate ticker, we need use the main ticker where gRPC apis are called periodically.
1. RetrieveActivationDetails() will return activate command until the current state is set to PROVISIONED. PMA should first check if activation is already triggered or not by checking the `amtinfo` output.
     i. If not triggered then trigger the `rpc activate...` and RepactActionvationResults as ACTIVATING.
    ii. If it already retriggered then parse the output and check for the connected string then RepactActionvationResults as ACTIVATED
    iii. If it is connecting then RepactActionvationResults as ACTIVATING.
    iv. if `rpc activate` command exit the non-zero then RepactActionvationResults as ACTIVATION_FAILED